homework/src/main/java/homework/exceptions/BookRepository.java

```
@@ -0,0 +1,55 @@
+package homework.exceptions;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import homework.exceptions.Book;
+
+public class BookRepository implements Repository<Book> {
+    Map<String, Book> repository = new HashMap<>();
+
+    @Override
+    public List findAll() {
+        return (List<String>) new ArrayList(repository.keySet());
+    }
+
+    @Override
+    public Book findById(String id) throws ItemNotFoundException {
+        Book book = null;
+        for (Map.Entry<String, Book> entry :
+                repository.entrySet()) {
+            if (entry.getValue().getId().equals(id)) {
+                book = entry.getValue();
+            }
+        }
+        if (book == null) {
```

you don't need to check on null.
if book with your id is present get or else throw Item not found exception. try to think about algorithm